pennylane_lightning/src/util/Kokkos_Sparse.hpp

```diff
@@ -0,0 +1,179 @@
+// Copyright 2022 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ * Contains Kokkos Sparse (linear algebra) utility functions.
+ */
+
+#pragma once
+
+#ifdef __has_include
+#if __has_include(                                                             \
+    <Kokkos_Core.hpp>) && __has_include (<KokkosKernels_default_types.hpp>)
+#include "KokkosKernels_default_types.hpp"
+#include "KokkosSparse_CrsMatrix.hpp"
+#include "KokkosSparse_spmv.hpp"
+#include "Kokkos_Core.hpp"
+#define _ENABLE_KOKKOS
+#endif
+#endif
+
+#ifdef _ENABLE_KOKKOS
+// Implementing Kokkos Sparse operations.
+#include <complex>
+#include <vector>
+
+namespace Pennylane::Util {
+using device_type = typename Kokkos::Device<
+    Kokkos::DefaultExecutionSpace,
+    typename Kokkos::DefaultExecutionSpace::memory_space>;
+
+using index_type = int;
```

Comment:
Kokkos prefers `int` for indices?

Reply:
Internally Kokkos performs loops with int. If kept as unsigned int, or size_t, a compilation performed with warnings as errors will fail.